tests/std/tests/P0896R4_ranges_alg_generate/test.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <range_algorithm_support.hpp>
+#include <ranges>
+#include <utility>
+
+constexpr void smoke_test() {
+    using ranges::generate, ranges::iterator_t;
+    using std::same_as;
+
+    {
+        int output[]    = {13, 42, 1367};
+        const int value = 7;
+        auto result     = generate(std::begin(output), std::end(output), []() { return value; });
+        for (auto elem : output) {
+            assert(elem == value);
+        }
+        assert(result == std::end(output));
+    }
+    {
+        int output[]    = {13, 42, 1367};
+        const int value = 13;
+        auto result     = generate(output, []() { return value; });
+        for (auto elem : output) {
+            assert(elem == value);
+        }
+        assert(result == std::end(output));
+    }
+    {
+        int output[]          = {13, 42, 1367};
+        int calls_to_generate = -1;
+        auto result           = generate(output, [&calls_to_generate]() {
+            calls_to_generate++;
+            return calls_to_generate;
+        });
+        for (int i = 0; i < 3; i++) {

```

Comment:
```suggestion
        for (int i = 0; i < 3; ++i) {
```
Just to beat STL to the comment 😄